### ISSUE TITLE
Inventory Setups Not Saving

### ISSUE THREAD

**Issue by User1893** [dillydill123/inventory-setups#206]
This issue started a week or two ago where whenever I add a new inventory setup or edit an old one, those changes are erased when I close the Runelite client and re-open it. So if I add a new setup and then close and re-open the client, that setup will have disappeared. If I edit any older setups that I made in the past and then close the client and re-open the client, those edits will have been erased and the setup will appear as it was a week or 2 ago (whenever this issue started)

My best guess for when this issue started is around 5/14/2023, as that is when I first made edits to a setup that I realized (a few days later) did not save. All the edits/new setups I had made since then were reverted, and this issue persists today. The only change/action I did around 5/14 that I can think of which maybe caused this was switching the plugin view from "standard mode" to "icon mode."

Otherwise, I use the inventory setups plugin pretty heavily, with 84 setups saved for almost every boss in the game and some skilling methods/mini-games. I also type out "summary guides" in the notes sections for tips and tricks and other things I've learned for future reference, so many of those inventory setups have large walls of text in their notes. Is it possible that my inventory setups plugin is "full" due to all that data, and that's why changes keep reverting?

Other Notes:
- I have not and am not signed into the same account on multiple Runelite clients.
- Uninstalling/re-installing Runelite does not resolve the issue, nor does logging in/out of Runelite profile.
- I'm on a Macbook.
- Any settings I change in my other Runelite plugins save whenever I close the client, so this issue seems to be unique to the inventory setups plugin.

Any help would be much appreciated.

**Comment by User1893** [comment]
Edit to the rough date the issue started: 4/14/2023, not 5/14/2023.

**Comment by dillydill123** [comment]
Can you supply your client.log file and your settings.properties? Make sure it's the one from your profile. If you don't want to share the whole thing, you can just supply the line `inventorysetups.setupsV2`. This was an issue in the past but I optimized the data storage **significantly** so that over 300 setups without notes should be possible. 

One way to check that it is this issue again is check the length of the line `inventorysetups.setupsV2` is close to 262144 characters. This is the RuneLite limit so if it goes over then it could be the cause.

**Comment by User1893** [comment]
Thanks for the quick response, I really appreciate it! I've linked the file that I believe you are referring to below - I'm on a Mac and don't know anything about computers, so let me know if you need something else. Thanks again for your help.

[Runelite Settings.txt](https://github.com/dillydill123/inventory-setups/files/11299962/Runelite.Settings.txt)

**Comment by dillydill123** [comment]
I don't see an `inventorysetups.setupsV2` in this settings. Are you sure it's the correct one? It should be in a folder in `.runelite/profiles` if you are logged into a RuneLite account. Also, find client.log in the `.runelite/logs`

**Comment by User1893** [comment]
Sorry, I think I found the actual files, see below. Regarding the client log file, since I tried uninstalling/re-installing earlier today, it looks like everything I did in the past was wiped. So I added "test setups" in runelite just to get the log to show what the error message is, hopefully that suffices for you.

[Settings Properties TXT.txt](https://github.com/dillydill123/inventory-setups/files/11300114/Settings.Properties.TXT.txt)

[Client Log TXT.txt](https://github.com/dillydill123/inventory-setups/files/11300115/Client.Log.TXT.txt)

**Comment by User1893** [comment]
Just realized all my settings are actually being saved in a different folder called profiles2 in a document called default-0.properties. This document shows my up-to-date settings. The document I linked above called settings.properties is from the regular profiles folder, but it hasn't been updated since February. 

[Settings Properties V2 TXT.txt](https://github.com/dillydill123/inventory-setups/files/11300218/Settings.Properties.V2.TXT.txt)

**Comment by dillydill123** [comment]
Got it, it looks like the settings.properties changed locations since the profile update happened. Taking a look, you are in fact going over the line limit, meaning that there's too much data to store.

I do apologize, since this issue is happening due to a design decision I made when I first created inventory setups. I think it's possible to fix, but it's not trivial and could jeopardize players data should I make a mistake. It also would take some time to do. I'll have to revisit the topic and see if it's worth the time.

RL could increase this line limit, but I don't think they want to go this route. So solution you will have to take is to reduce the amount of notes in your setups 😢 I know it's not the answer you would like, but it's the only solution I can give right now.

For reference, here is the limit as decided by RL: 
https://github.com/runelite/api.runelite.net/blob/f1d34a17263c37d09d96eb51b1736db6bc8d8768/http-service/src/main/java/net/runelite/http/service/config/ConfigService.java#L74

**Comment by User1893** [comment]
No worries, I really appreciate your help on this and for creating this plugin in general! I wasn't lying about the walls of text... Pretty easy to just save some of those notes elsewhere so I'm actually glad that this is the issue as opposed to some other issue where I'd have to delete/re-create all my setups.

**Comment by KillerEnraged** [comment]
Hey Dilly, I am experiencing this same exact issue with some minor differences. My line length is not at max (looks like im around 93k characters long). My setups are saving upon closing however, if I do not remember to close the client before runelite updates it seems as if my setups are backdated. Any changes I made or any new setups I made get reverted. I am not sure precisely when it started happening, but the date seems to align with this issue. I am not certain how far the back date goes (if its the last time I closed the client or a further back date)
